src/tests/ascent/t_ascent_commands.cpp

```diff
@@ -0,0 +1,408 @@
+TEST(ascent_commands, register_no_name_bool)
+{
+    // Reset callbacks
+    ascent::reset_callbacks();
+
+    // An error should be thrown due to not including a name
+    EXPECT_ANY_THROW(ascent::register_callback("", bool_callback_1));
+}
+
+//-----------------------------------------------------------------------------
+TEST(ascent_commands, register_void_callbacks)
+{
+    // Reset callbacks
+    ascent::reset_callbacks();
+
+    // No error should be thrown
+    EXPECT_NO_THROW(ascent::register_callback("void_callback_1", void_callback_1));
+    EXPECT_NO_THROW(ascent::register_callback("void_callback_2", void_callback_2));
+}
+
+//-----------------------------------------------------------------------------
+TEST(ascent_commands, register_bool_callbacks)
+{
+    // Reset callbacks
+    ascent::reset_callbacks();
+
+    // No error should be thrown
+    EXPECT_NO_THROW(ascent::register_callback("bool_callback_1", bool_callback_1));
+    EXPECT_NO_THROW(ascent::register_callback("bool_callback_2", bool_callback_2));
+}
+
+//-----------------------------------------------------------------------------
+TEST(ascent_commands, register_same_callback_twice_mixed)
+{
+    // Reset callbacks
+    ascent::reset_callbacks();
+
+    // Register callbacks
+    EXPECT_NO_THROW(ascent::register_callback("test_callback", void_callback_1));
+
+    // An error should be thrown for registering the same callback name twice
+    EXPECT_ANY_THROW(ascent::register_callback("test_callback", bool_callback_1));
+}
+
+//-----------------------------------------------------------------------------
+TEST(ascent_commands, register_same_void_callback_twice)
+{
+    // Reset callbacks
+    ascent::reset_callbacks();
+
+    // Register callbacks
+    EXPECT_NO_THROW(ascent::register_callback("void_callback", void_callback_1));
+
+    // An error should be thrown registering the same callback name twice
+    EXPECT_ANY_THROW(ascent::register_callback("void_callback", void_callback_2));
+}
+
+//-----------------------------------------------------------------------------
+TEST(ascent_commands, register_same_bool_callback_twice)
+{
+    // Reset callbacks
+    ascent::reset_callbacks();
+
+    // Register callbacks
+    EXPECT_NO_THROW(ascent::register_callback("bool_callback", bool_callback_1));
+
+    // An error should be thrown registering the same callback name twice
+    EXPECT_ANY_THROW(ascent::register_callback("bool_callback", bool_callback_2));
+}
+
+//-----------------------------------------------------------------------------
+TEST(ascent_commands, direct_void_callback_invocation)
+{
+    // Reset callbacks
+    ascent::reset_callbacks();
+
+    // Register callbacks
+    ascent::register_callback("void_callback_1", void_callback_1);
+
+    //
+    // Run Ascent
+    //
+    Ascent ascent;
+    Node ascent_opts;
+    // default is now ascent
+    ascent_opts["runtime/type"] = "ascent";
+    ascent.open(ascent_opts);
```

Comment:
for this test case we aren't actually using the ascent instance, so we can remove creating it.